internal/service/dataexchange/revision_assets.go

```diff
@@ -0,0 +1,1085 @@
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataExchange, create.ErrActionReading, ResNameRevisionAssets, state.ID.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, out, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	setTagsOut(ctx, out.Tags)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceRevisionAssets) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	conn := r.Meta().DataExchangeClient(ctx)
+
+	var plan, state resourceRevisionAssetsModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	diff, d := flex.Diff(ctx, plan, state, flex.WithIgnoredField("ForceDestroy"))
+	resp.Diagnostics.Append(d...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if diff.HasChanges() {
+		input := dataexchange.UpdateRevisionInput{
+			DataSetId:  plan.DataSetID.ValueStringPointer(),
+			RevisionId: plan.ID.ValueStringPointer(),
+		}
+		resp.Diagnostics.Append(flex.Expand(ctx, plan, &input)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+
+		if !plan.Comment.Equal(state.Comment) && state.Finalized.ValueBool() {
+			err := finalizeAsset(ctx, conn, plan.DataSetID.ValueString(), plan.ID.ValueString(), false)
+			if err != nil {
+				resp.Diagnostics.AddError(
+					create.ProblemStandardMessage(names.DataExchange, create.ErrActionUpdating, ResNameRevisionAssets, plan.ID.String(), err),
+					err.Error(),
+				)
+				return
+			}
+		}
+
+		out, err := conn.UpdateRevision(ctx, &input)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.DataExchange, create.ErrActionUpdating, ResNameRevisionAssets, plan.ID.String(), err),
+				err.Error(),
+			)
+			return
+		}
+
+		resp.Diagnostics.Append(flex.Flatten(ctx, out, &plan)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *resourceRevisionAssets) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	conn := r.Meta().DataExchangeClient(ctx)
+
+	var state resourceRevisionAssetsModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if state.ForceDestroy.ValueBool() && state.Finalized.ValueBool() {
+		err := finalizeAsset(ctx, conn, state.DataSetID.ValueString(), state.ID.ValueString(), false)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.DataExchange, create.ErrActionDeleting, ResNameRevisionAssets, state.ID.String(), err),
+				err.Error(),
+			)
+			return
+		}
+	}
+
+	input := dataexchange.DeleteRevisionInput{
+		DataSetId:  state.DataSetID.ValueStringPointer(),
+		RevisionId: state.ID.ValueStringPointer(),
+	}
+
+	_, err := conn.DeleteRevision(ctx, &input)
```

Comment:
We might want to consider a callout on the registry docs that indicates destruction will destroy both the assets and underlying revision. Given the resource name, users might expect destroy to remove associated assets and leave the underlying revision, especially since we have a separate resource for just the revision. 

On this topic, would it make sense to deprecate `aws_dataexchange_revision` given it doesn't provide any utility without assets?

Reply:
@jar-b we can add a deprecation in a separate PR. It would make sense to deprecate but I almost feel like we need to support import first

Reply:
Makes sense, I forgot about the import complications since the last review. Will close this out in favor of a follow-up issue.